cmd/config/notify/parse.go

```diff
@@ -307,17 +356,16 @@ var (
 )
 
 // GetNotifyKafka - returns a map of registered notification 'kafka' targets
-func GetNotifyKafka(s config.Config) map[string]target.KafkaArgs {
+func GetNotifyKafka(kafkaKVS map[string]config.KVS) (map[string]target.KafkaArgs, error) {
 	kafkaTargets := make(map[string]target.KafkaArgs)
-	for k, kv := range mergeTargets(s[config.NotifyKafkaSubSys], target.EnvKafkaState, DefaultKafkaKVS) {
+	for k, kv := range mergeTargets(kafkaKVS, target.EnvKafkaState, DefaultKafkaKVS) {
```

Comment:
mergeTargets will not pick the env correctly imo, because env.List uses EnvKafkaState as a prefix. Instead MINIO_NOTIFY_KAFKA_ should be passed to mergeTargets. Same comment for all other notification targets.

Reply:
No it does @poornas the state is used as a way to check what is enabled @poornas the envs are further constructed after the fact that _STATE is available

Reply:
mergeTargets is only used to obtain either its Default or custom target